common/core/src/retry_policy.ts

```diff
@@ -156,6 +159,10 @@ export class ExponentialBackOffWithJitter implements RetryPolicy {
  * @implements {RetryPolicy}
  */
 export class NoRetry implements RetryPolicy {
+  constructor() {
```

Comment:
Should this be in the constructor for RetryOperation?

Reply:
Someting like policy.constructor.name === 'NoRetry'

Reply:
Wouldn't it cause a problem if the user creates a custom retry policy called `NoRetry`? Maybe this case is insignificant enough to do it the way you suggest.

Reply:
Well, what we are doing here can NOT affect the running of the application SDK code.  In the end all we are producing is a debug.

Second, if they hand us a retry policy whose constructor is named NoRetry, I'm going to assume that it doesnt retry.

The nice thing here is as opposed to simply instantiating something that is NoRetryish, but moving this to the RetryOperation, we can say the developer is actually doing utilizing something that is NoRetryish.